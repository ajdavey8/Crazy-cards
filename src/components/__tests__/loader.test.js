import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Loader from '../loader';

describe('Loader', () => {
  const wrapper = shallow(<Loader />);

  it('renders and matches snapshot', () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the loading spinner', () => {
    expect(wrapper.find('i.fa-spinner').exists()).toBe(true);
  });

  it('renders the loading text', () => {
    expect(wrapper.find('.loader__text').text()).toContain('Loading...');
  });
});
