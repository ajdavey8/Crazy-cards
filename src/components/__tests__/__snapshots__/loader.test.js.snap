// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders and matches snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="loader__icon"
  >
    <i
      className="fas fa-spinner fa-pulse fa-4x fa-fw"
    />
  </div>
  <h3
    className="loader__text"
  >
    Loading...
  </h3>
</div>
`;
